lib/src/network/url_tools.dart

```diff
@@ -0,0 +1,53 @@
+part of '../mistralai_client_dart_base.dart';
```

Comment:
Does it have to be a part of the base file? 🤔 It seems like just a file with some public classes 🤔

Reply:
By doing it like that, if you want to override `apiUrlFactory` you don't have to import additional files. You get all the public functionalities of our package by importing one file.

Reply:
This one doesn't have to. We could make it a proper separate file and export it.

Reply:
Refactored in [7eb533e](https://github.com/nomtek/mistralai_client_dart/pull/18/commits/7eb533ed64440dd2129a974385b1c8efc0fc740d)